### PR TITLE
update testng to 7.3.0

### PR DESCRIPTION
**Pull request by xinlian12**
Fix issue: https://github.com/Azure/azure-sdk-for-java/issues/21777

The JCommander:1.72 is a dependency for org.testng:testing:6.14.3: https://mvnrepository.com/artifact/org.testng/testng/6.14.3

image

Testing 7.3 and Testing 7.4 both have dependency on JCommander 1.78. But can not upgrade to 7.4 for now, because of bug in testing 7.4:
https://github.com/cbeust/testng/blob/master/CHANGES.txt
Fixed: GITHUB-2493: Avoid NPE from TextReporter execution when a dataprovider method provides null

Also need to update to use https to load testing did due to:https://github.com/cbeust/testng/pull/2023

Note: After upgrading to 7.3, you might experience the following exception when running tests with Intellij: "TestNG by default disables loading DTD from unsecure Urls". 
Two ways to solve this issue:
1. Add the following VM option: -Dtestng.dtd.http=true
2. Upgrade IntelliJ to a newer version, tested working version: https://www.jetbrains.com/idea/whatsnew/#section=mac

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-benchmark/pom.xml
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/pom.xml:131]: testng is deeply used for running all integration tests. If we are upgrading testng version we should run the full test CI and make sure we have a clean Ci result. before merging.

**Review by moderakh — COMMENTED** [review]
testng is deeply used for running all integration tests. If we are upgrading testng version we should run the full test CI and make sure we have a clean Ci result. before merging.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java:554]: Where is this used ?

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java:554]: in the changeFeedTest.java

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java:554]: @Test(groups = { "emulator" }, enabled = false, timeOut = TIMEOUT)
    @Tag(name = "EnableFullFidelity")
    public void changeFeed_fullFidelity_fromNow() throws Exception {
        changeFeed_withUpdatesAndDelete(true);
    }

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - thanks

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java:552]: why are these needed in this PR?

are these features removed from testng? and we are introducing our own?

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java:552]: yes. originally we are using Tag from here: https://javadoc.io/doc/org.testng/testng/7.1.0/org/testng/xml/dom/package-summary.html

but from 7.3.0, I can not find this package any more, so created a new one

Comment (moderakh) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java:552]: I see. are we currently using that testng tag for grouping tests or running a subset of tests?

how are we planning to wire those test groups against the new tag. any work needed?

Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ChangeFeedTest.java:552]: Currently Tag is only used in the changeFeedTests. It is not used as group tests etc. No other work is needed.

Currently I think  it mainly used as a custom attributes, which is used to differentiate which document is created: 

@BeforeMethod(groups = { "simple", "emulator" }, timeOut = SETUP_TIMEOUT)
    public void populateDocuments(Method method) {

        checkNotNull(method, "Argument method must not be null.");

        Tag tag = method.getDeclaredAnnotation(Tag.class);
        if (tag != null && "EnableFullFidelity".equalsIgnoreCase(tag.name())) {
            populateDocumentsInternal(method, true);
        } else {
            populateDocumentsInternal(method, false);
        }
    }

**Review by moderakh — CHANGES_REQUESTED** [review]
please don't merge yet. we have some test infra issue which causes not running all tests.

**Review by moderakh — APPROVED** [review]
LGTM. thanks

**Pull request merged**
